src/battle_script_commands.c

```diff
@@ -8121,8 +8121,10 @@ static bool32 HasAttackerFaintedTarget(void)
 
 bool32 CanPoisonType(u8 battlerAttacker, u8 battlerTarget)
 {
-    return ((GetBattlerAbility(battlerAttacker) == ABILITY_CORROSION && gBattleMoves[gCurrentMove].split == SPLIT_STATUS)
-            || !(IS_BATTLER_OF_TYPE(battlerTarget, TYPE_POISON) || IS_BATTLER_OF_TYPE(battlerTarget, TYPE_STEEL)));
+    if (GetBattlerAbility(battlerAttacker) != ABILITY_CORROSION
```

Comment:
Last change, I promise haha. Use de morgan on this and just return the statement directly like it was originally:
```
    return GetBattlerAbility(battlerAttacker) == ABILITY_CORROSION
        || (!IS_BATTLER_OF_TYPE(battlerTarget, TYPE_STEEL) && !IS_BATTLER_OF_TYPE(battlerTarget, TYPE_POISON))
```

Reply:
it's fine. Hope you aren't getting tired by my coding abilities 😅

Reply:
pushed the change.